### PR TITLE
Fix regex for proper target recognition

### PR DESCRIPTION
**Pull request by mzfr**
#### Checklist
- [x] I have followed the [Contributor Guidelines](https://github.com/zdresearch/OWASP-Nettacker/wiki/Developers#contribution-guidelines).
- [x] I have added the relevant documentation.
- [x] My branch is up-to-date with the Upstream master branch.

#### Changes proposed in this pull request

Issue: #185 
This PR fixes the bug due to which targets were not properly distinguished.

#### Your development environment
- OS: `Manjaro`
- OS Version: `4.20`
- Python Version: `3.5`

### CODE REVIEW

File: core/targets.py [hunk 1]
```diff
@@ -47,11 +47,11 @@ def target_type(target):
         start_ip, stop_ip = target.rsplit('-')
         if isIP(start_ip) and isIP(stop_ip):
             return 'RANGE_IPv4'
-    elif re.match('^([a-z0-9]+(-[a-z0-9]+)*\.)+[a-z]{2,}$', target):
+    elif re.match('^([a-zA-Z0-9]+(-|_[a-zA-Z0-9]+)*\.?)+[a-z]{2,}$', target):
```
Comment (shaddygarg) [inline core/targets.py]:
```suggestion
    elif re.match('^([a-zA-Z0-9]+(-|_[a-zA-Z0-9]+)*\.?)+[a-zA-Z]{2,}$', target):
```

File: core/targets.py [hunk 2]
```diff
@@ -47,11 +47,11 @@ def target_type(target):
...
         return 'DOMAIN'
     elif (target.lower().startswith('http://') or target.lower().startswith('https://')):
         t = target.rsplit("://")[1].rsplit("/")[0].rsplit(":")[0]
-        if isIP(t) or isIP6(t) or re.match('^([a-z0-9]+(-[a-z0-9]+)*\.)+[a-z]{2,}$', t):
+        if isIP(t) or isIP6(t) or re.match('^([a-zA-Z0-9]+(-|_[a-zA-Z0-9]+)*\.?)+[a-z]{2,}$', t):
```
Comment (shaddygarg) [inline core/targets.py]:
```suggestion
        if isIP(t) or isIP6(t) or re.match('^([a-zA-Z0-9]+(-|_[a-zA-Z0-9]+)*\.?)+[a-zA-Z]{2,}$', t):
```

**Comment by mzfr** [comment]
@shaddygarg I have made the suggested changes

**Comment by mzfr** [comment]
@shaddygarg @Ali-Razmjoo What changes are required in this?

**Comment by shaddygarg** [comment]
I think it is good to go. :+1:

**Comment by shaddygarg** [comment]
@Ali-Razmjoo this can be merged.

**Comment by Ali-Razmjoo** [comment]
thanks for contribution @mzfr , @shaddygarg thanks for review.

**Pull request merged**